clippy_lints/src/ranges.rs

```diff
@@ -292,7 +294,10 @@ fn check_possible_range_contains(cx: &LateContext<'_>, op: BinOpKind, l: &Expr<'
     }
 }
 
-fn check_range_bounds(cx: &LateContext<'_>, ex: &Expr<'_>) -> Option<(Constant, HirId, Span, Span, Ordering, bool)> {
+fn check_range_bounds<'a>(
+    cx: &'a LateContext<'_>,
+    ex: &'a Expr<'_>,
+) -> Option<(Constant, &'a Expr<'a>, HirId, Span, Span, Ordering, bool)> {
```

Comment:
NIT: This return type is very complex and not descriptive. Would you mind creating a struct with named values instead?

If not, we can also create an issue to this :upside_down_face:

Reply:
Thanks! I added a struct, how does it look?